parser/src/config/lexer.rs

```diff
@@ -206,4 +207,93 @@ mod test {
 
         assert_eq!(lexed, expected);
     }
+
+    #[test]
+    fn error_location_byte_offset_conversion() {
+        let test_cases = vec![
+            // (text, offset, expected_line, expected_column)
+            ("property: value", 0, 1, 1),              // Start of text
+            ("property: value", 8, 1, 9),              // At colon
+            ("property: value", 10, 1, 11),            // After space and before 'v'
+            ("name: \"test\"\ncolor: blue", 13, 2, 1), // Start of line 2 (offset after newline)
+            ("name: \"test\"\ncolor: blue", 19, 2, 7), // At space before 'blue'
+            ("a: b\nc: d\ne: f", 5, 2, 1), // Start of line 2 (offset after first newline)
+            ("a: b\nc: d\ne: f", 10, 3, 1), // Start of line 3 (offset after second newline)
```

Comment:
Just for consistency. ✨ 

```suggestion
            ("name: \"test\"\ncolor: blue", 19, 2, 7), // At space before 'blue'
            ("a: b\nc: d\ne: f", 5, 2, 1),             // Start of line 2 (offset after first newline)
            ("a: b\nc: d\ne: f", 10, 3, 1),            // Start of line 3 (offset after second newline)
```